packages/sanity/src/core/store/_legacy/document/document-pair/checkoutPair.ts

```diff
@@ -79,19 +79,27 @@ function requireId<T extends {_id?: string; _type: string}>(
   }
 }
 
+//if we're patching a published document directly
+//then we're live editing and we should use raw mutations
+//rather than actions
+function isLiveEditMutation(mutationParams: Mutation['params'], publishedId: string) {
+  const {resultRev, ...mutation} = mutationParams
+  const patchTargets: string[] = mutation.mutations.flatMap((mut) => {
+    const mutationPayloads = Object.values(mut)
+    if (mutationPayloads.length > 1) {
+      throw new Error('Did not expect multiple mutations in the same payload')
+    }
+    return mutationPayloads[0].id || mutationPayloads[0]._id
+  })
+  return patchTargets.every((target) => target === publishedId)
+}
+
 function toActions(idPair: IdPair, mutationParams: Mutation['params']) {
   return mutationParams.mutations.flatMap<HttpAction>((mutations) => {
     if (Object.keys(mutations).length > 1) {
       // todo: this might be a bit too strict, but I'm (lazily) trying to check if we ever get more than one mutation in a payload
       throw new Error('Did not expect multiple mutations in the same payload')
     }
-    if (mutations.delete) {
```

Comment:
I did not find anywhere in the codebase where `draft.mutate([draft.delete()])` occurred. 

In the interest of not having unused codepaths, I deleted this. If we're more concerned about completion we can certainly add it back -- I was worried about making this file too noisy.

Reply:
Good call! We have never done deletes optimistically (via buffered documents), so this makes sense to me.